### PR TITLE
Add vm.unixTime() cheatcode

### LINKED ISSUE #5937: Feature request: vm.time() cheatcode to return current system time in milliseconds since epoch
### Component

Forge

### Describe the feature you would like

It would be nice to have a cheatcode for getting the current system time in milliseconds since epoch.

Personally my main use case is for measuring performance of scripts. Some scripts can become quite complex and I encounter performance issues and I find it useful to measure the time it takes certain parts of the script to run.

Currently my solution is to use `vm.ffi(['python', '-c', 'import time; print(int(time.time() * 1000))'])` :sweat_smile: 

The interface I have in mind is:
```solidity
function time () view returns (uint millisecondsSinceEpoch);
```

I'll also happily open a PR if needed :slightly_smiling_face: 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Tudmotu**
## Motivation
Fixes #5937.
For long-running scripts it is sometimes useful to measure certain parts and analyze their performance in order to optimize running time. `block.timestamp` does not provide a real-time system timestamp, and using `date` via `ffi` is not granular enough due to Mac OS lack of milliseconds support.

This feature will return an accurate timestamp, in milliseconds since unix epoch, as `uint`.

## Solution
I've added the usual cheatcode components:
- Added function to `HEVM.sol`
- Generated abi bindings
- Added implementation to `foundry-evm` (specifically, `ext.rs`)
- Added Solidity tests in `testdata`

Implementation details:
- Using `SystemTime::now().duration_since(UNIX_EPOCH)`, following the [example from the docs](https://doc.rust-lang.org/std/time/struct.SystemTime.html#impl-SystemTime)
- Converting `u128` to `Bytes` via `rU256` 

## Open questions
- The following test fails both on my branch and on master: `'otterscan::can_call_ots_has_code'`.
Is this ok?
- Using`rU256` to convert the `u128` from `.as_millis()` to bytes — is there a nicer way to achieve this?
- Should this be renamed to something else, such as `.unixTime()`, `.systemTime()` or `.timestamp()`? Personally I would have preferred `vm.now()` but `now` is a reserved word in Solidity

### CODE REVIEW

File: crates/evm/src/executor/inspector/cheatcodes/ext.rs
Comment (mattsse) [inline crates/evm/src/executor/inspector/cheatcodes/ext.rs:521]: let's call this duration_since_epoch

File: crates/evm/src/executor/inspector/cheatcodes/ext.rs
Comment (mattsse) [inline crates/evm/src/executor/inspector/cheatcodes/ext.rs:527]: we need to encode this, types are currently a bit horrible until we completely transitioned to alloy types we need to use:
`Ok(rU256::from(millis).to_ethers().encode().into())`

**Review by mattsse — CHANGES_REQUESTED** [review]
cool, I like this,

smol nits

**Comment by Tudmotu** [comment]
Thanks @mattsse for the review, will make the changes and update :slightly_smiling_face:

**Comment by Tudmotu** [comment]
@mattsse I have addressed your comments and also went ahead and changed the cheatcode to `unixTime()`.
Can still modify it if there's any objection to that name.

**Review by Evalir — APPROVED** [review]
personally don't have typos—as long as this test passes I think this is good! deferring to @mattsse & @mds1 for merge (also needs `cargo +nightly fmt`)

**Comment by Tudmotu** [comment]
Apologies, I messed up the format in the last change and forgot to fix it.
Pushed a fix now.

**Review by Evalir — APPROVED** [review]
let's remember to update the book & forge-std once this is merged!

**Comment by Tudmotu** [comment]
@Evalir I already got foundry book & forge-std updates prepared, just waiting for this to be merged before I open PRs :slightly_smiling_face:

**Review by mattsse — APPROVED** [review]
lgtm, tysm!

**Comment by Tudmotu** [comment]
Thank you all, you're doing amazing work with Foundry. And I must add the PR process was very smooth, well done! :clap:

**Pull request merged**
